File: app/src/androidTest/java/com/glsebastiany/bakingapp/core/MockApplication.java
package com.glsebastiany.bakingapp.core;

import com.glsebastiany.bakingapp.MyApplication;
import com.glsebastiany.bakingapp.core.injection.MockedApplicationModule;
import com.glsebastiany.bakingapp.injection.component.ApplicationComponent;
import com.glsebastiany.bakingapp.injection.component.DaggerApplicationComponent;

public class MockApplication extends MyApplication {

    @Override
    public ApplicationComponent buildApplicationComponent() {
        return DaggerApplicationComponent
                .builder()
                .applicationModule(new MockedApplicationModule(this))
                .build();
    }

}

File: app/src/androidTest/java/com/glsebastiany/bakingapp/core/MockApplication.kt
package com.glsebastiany.bakingapp.core

import com.glsebastiany.bakingapp.MyApplication
import com.glsebastiany.bakingapp.core.injection.MockedApplicationModule
import com.glsebastiany.bakingapp.injection.component.ApplicationComponent
import com.glsebastiany.bakingapp.injection.component.DaggerApplicationComponent

class MockApplication : MyApplication() {

    override fun buildApplicationComponent(): ApplicationComponent {
        return DaggerApplicationComponent
                .builder()
                .applicationModule(MockedApplicationModule(this))
                .build()
    }
}
